### PR TITLE
Release 0.1.77

### PR DESCRIPTION
**Pull request by corysquid**
- Trader Joe v2.1 types

* chore: add Beamswap Saddle DexName

* chore: add Lynex V3 DexName

* fix: Pharaoh Solidly DexName

* chore: Nile/Ramses Solidly DexName

- Add back Ramses Solidly
- Add Nile Solidly (Linea)

* chore: add Ra, Pharaoh and Nile V3 DexNames

### CODE REVIEW

**Review by ecdsafu — APPROVED** [review]
Names look good

File: src/dexes/index.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export enum DexName {
   BASESWAP_V3 = "Baseswap V3",
   BEAMSWAP_V2 = "Beamswap V2",
   BEAMSWAP_V3 = "Beamswap V3",
+  BEAMSWAP_SADDLE = "Beamswap Saddle",
```
Comment (0x0Koda) [inline src/dexes/index.ts:21]: @ecdsafu do you want the dex type in the name Beamswap Saddle" or beamswaps terminology"Beamswap V2 Stable"

File: src/dexes/index.ts [hunk 2]
```diff
@@ -31,17 +32,23 @@ export enum DexName {
...
   MENTO_V2 = "Mento V2",
+  NILE_SOLIDLY = "Nile Solidly",
+  NILE_V3 = "Nile V3",
   OPENOCEAN = "OpenOcean",
   OSMOSIS = "Osmosis",
   PANCAKESWAP_V2 = "Pancakeswap V2",
   PANCAKESWAP_V3 = "Pancakeswap V3",
   PANCAKESWAP_STABLE = "Pancakeswap Stable",
   PANGOLIN_V2 = "Pangolin V2",
+  PHARAOH_SOLIDLY = "Pharaoh Solidly",
```
Comment (0x0Koda) [inline src/dexes/index.ts:45]: Same comment as Beamswap

File: src/dexes/index.ts [hunk 3]
```diff
@@ -31,17 +32,23 @@ export enum DexName {
   KYBERSWAP = "KyberSwap",
   KYBERSWAP_AGGREGATOR = "kyberswap Aggregator",
   KINETIX_V3 = "Kinetix V3",
+  LYNEX_V3 = "Lynex V3",
   MENTO_V2 = "Mento V2",
+  NILE_SOLIDLY = "Nile Solidly",
```
Comment (0x0Koda) [inline src/dexes/index.ts:37]: Same comment as Beamswap

**Review by 0x0Koda — COMMENTED** [review]
comments for @ecdsafu

**Review by 0x0Koda — APPROVED** [review]

File: src/dexes/index.ts (hunk 1 above)
Comment (ecdsafu) [reply src/dexes/index.ts:21]: It's a good idea, maybe the branded terminology is the way to go.

**Pull request merged**